### PR TITLE
[PATCH 2.0 v3] allocate ring memory based on pool size

### PR DESCRIPTION
**Pull request by nagarahalli**
The maximum size of the pool ring is 1M entries as it uses an array. However, the application provides the number of buffers in a pool during pool creation. Hence, the maximum pool ring size can be dynamic.

### CODE REVIEW

File: platform/linux-generic/pool/generic.c [hunk 1]
```diff
@@ -506,11 +510,6 @@ static int check_params(odp_pool_param_t *params)
 
 	switch (params->type) {
 	case ODP_POOL_BUFFER:
-		if (params->buf.num > capa.buf.max_num) {
```
Comment (Bill-Fischofer-Linaro) [inline platform/linux-generic/pool/generic.c]: Same here:
```
if (capa.buf.max_num && params->buf.num > capa.buf.max_num) ...
```

File: platform/linux-generic/pool/generic.c [hunk 2]
```diff
@@ -550,10 +549,6 @@ static int check_params(odp_pool_param_t *params)
 		break;
 
 	case ODP_POOL_TIMEOUT:
-		if (params->tmo.num > capa.tmo.max_num) {
```
Comment (Bill-Fischofer-Linaro) [inline platform/linux-generic/pool/generic.c]: And here too.

**Comment by Bill-Fischofer-Linaro** [comment]
We discussed this PR briefly in the Wednesday ARCH call. @psavol asked for the rationale behind it. Presumably it is to conserve memory? Can we quantify the expected savings?

**Comment by nagarahalli** [comment]
@Bill-Fischofer-Linaro odp_pool_create API already takes number of buffers/packets required in the pool as input from the user. For ODP 2.0, only restriction to support any number is available amount of memory. Hence, it is not required to add an artificial limit which is also an additional configuration to worry about.

May be it is ok to have that limit in Linux-Generic to show clarity, but for ODP 2.0, such a limitation is not required.

**Comment by nagarahalli** [comment]
@Bill-Fischofer-Linaro I suggest that we look into how this will go to master when Tiger Moth is done. For now, we should get it into ODP 2.0.

**Comment by heyi-linaro** [comment]
Merged and closed.

**Pull request closed without merging**